### PR TITLE
Show highlighted symbol in the scrollbar

### PR DESCRIPTION
**Pull request by fxsalazar**
Release Notes:

- Added highlighted symbols to the scrollbar; partially mentioned in:
  - https://github.com/zed-industries/zed/issues/5308
  - https://github.com/zed-industries/zed/issues/4866

### CODE REVIEW

**Comment by fxsalazar** [comment]
@cla-bot check'

**Review by mikayla-maki — APPROVED** [review]
Thank you for this addition!

File: crates/editor/src/element.rs [hunk 1]
```diff
@@ -1366,6 +1366,44 @@ impl EditorElement {
                 }
             }
 
+            if layout.is_singleton && scrollbar_settings.symbols_selections {
```
Comment (mikayla-maki) [inline crates/editor/src/element.rs:1369]: Above here, there's a check for 'show_scrollbars', that wasn't updated to incorporate this new information. Could you add a branch to resolution of the `ShowScrollbar::Auto` setting, on line 2066?

**Review by mikayla-maki — CHANGES_REQUESTED** [review]

File: crates/editor/src/element.rs [hunk 2]
```diff
@@ -2032,8 +2070,12 @@ impl EditorElement {
                     ||
                     // Selections
                     (is_singleton && scrollbar_settings.selections && editor.has_background_highlights::<BufferSearchHighlights>())
+                    ||
+                    // Symbols Selections
+                    (is_singleton && scrollbar_settings.symbols_selections && editor.has_background_highlights::<BufferSearchHighlights>())
```
Comment (mikayla-maki) [inline crates/editor/src/element.rs:2075]: I believe this should be `DocumentHighlightRead` or `DocumentHighlightWrite`, as shown in `Editor::refresh_document_highlights`

**Review by mikayla-maki — CHANGES_REQUESTED** [review]

**Review by mikayla-maki — APPROVED** [review]

**Comment by mikayla-maki** [comment]
Thank you!

**Pull request merged**